trie/utils/verkle.go

```diff
@@ -63,31 +63,34 @@ func GetTreeKey(address []byte, treeIndex *uint256.Int, subIndex byte) []byte {
 		var aligned [32]byte
 		address = append(aligned[:32-len(address)], address...)
 	}
-	var poly [5]fr.Element
 
-	poly[0].SetZero()
+	// poly = [2*64<<8, addressLE_low, addressLE_high, trieIndexLE_low, trieIndexLE_high]
+	var poly [5]fr.Element
 
-	// 32-byte address, interpreted as two little endian
-	// 16-byte numbers.
+	// 32-byte address, interpreted as two little endian 16-byte numbers.
 	verkle.FromLEBytes(&poly[1], address[:16])
 	verkle.FromLEBytes(&poly[2], address[16:])
 
-	// little-endian, 32-byte aligned treeIndex
-	var index [32]byte
-	for i, b := range treeIndex.Bytes() {
-		index[len(treeIndex.Bytes())-1-i] = b
-	}
-	verkle.FromLEBytes(&poly[3], index[:16])
-	verkle.FromLEBytes(&poly[4], index[16:])
-
-	cfg, _ := verkle.GetConfig()
+	// treeIndex must be interpreted as a 32-byte aligned little-endian integer.
+	// e.g: if treeIndex is 0xAABBCC, we need the byte representation to be 0xCCBBAA00...00.
+	// poly[3] = LE({CC,BB,AA,00...0}) (16 bytes), poly[4]=LE({00,00,...}) (16 bytes).
+	//
+	// To avoid unnecessary endianness conversions for go-ipa, we do some trick:
+	// - poly[3] integer is the same as the *top* 16 bytes (trieIndexBytes[16:]) of
+	//   32-byte aligned big-endian representation (BE({00,...,AA,BB,CC})).
+	// - poly[4] integer is the same as the *low* 16 bytes (trieIndexBytes[:16]) of
+	//   the 32-byte aligned big-endian representation (BE({00,00,...}).
+	trieIndexBytes := treeIndex.Bytes32()
+	verkle.FromBytes(&poly[3], trieIndexBytes[16:])
+	verkle.FromBytes(&poly[4], trieIndexBytes[:16])
```

Comment:
OK, so the new implementation of the section is quite short which is nice.

Despite that, what is doing can feel confusing mostly because the `[16:]` and `[:16]` seems to be shifted, if compared to L71 and L72; but isn't an error.

Honestly, I preferred to explain what is happening here. I usually prefer to be a bit verbose since this code can be tricky to understand again in the future for somebody else, and it is easy to think it might be wrong.

If feels too verbose or feels obvious enough, let me know and I can delete the comment.